### PR TITLE
Scale svg correctly

### LINKED ISSUE #728: Some svg look pixelated
### Actual behaviour
screenshot_20180223-174050

### Expected behaviour
Person icon shouldn't be so pixelated, because it's SVG.

 
### Steps to reproduce
1. Use this icon:
````xml
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" baseProfile="full" width="24" height="24" viewBox="0 0 24.00 24.00" enable-background="new 0 0 24.00 24.00" xml:space="preserve">
	<path fill="#E7B48F" fill-opacity="1" stroke-width="0.2" stroke-linejoin="round" d="M 12,2C 13.1046,2 14,2.89543 14,4C 14,5.10457 13.1046,6 12,6C 10.8954,6 10,5.10457 10,4C 10,2.89543 10.8954,2 12,2 Z M 10.5,7.00001L 13.5,7.00001C 14.6046,7.00001 15.5,7.89544 15.5,9.00001L 15.5,14.5L 14,14.5L 14,22L 10,22L 10,14.5L 8.5,14.5L 8.5,9.00001C 8.5,7.89544 9.39543,7.00001 10.5,7.00001 Z "/>
</svg>
````
Source: https://github.com/Templarian/MaterialDesign/blob/master/icons/svg/human-male.svg

2. Set icon format to svg.

### Can you reproduce the issue in demo mode?
No.

#### Client
* Android version: Android 7.1.2
* Device model: Oneplus X
* Stock or customized ROM: LineageOS 14.1
* App version : 2.0.15-beta
* Installation source : Github
* Device language: DE

#### Server
Client issue...

### PR DESCRIPTION
**Pull request by mueller-ma**
Fixes #728

@maniac103 I copied and modified your code from OctoDroid. How do I need to mention you or OctoDroid in the comments to avoid copyright violations?

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/util/Util.java [hunk 1]
```diff
@@ -271,4 +273,30 @@ public static void applySwipeLayoutColors(SwipeRefreshLayout swipeLayout,
...
+     *
+     * @param px A value in px (pixels) unit. Which we need to convert into db
+     * @param context Context to get resources and device specific display metrics
+     * @return A float value to represent dp equivalent to px value
+     * @author https://stackoverflow.com/a/9563438
+     */
+    public static int convertPixelsToDp(float px, Context context){
+        return Math.round(px / ((float) context.getResources().getDisplayMetrics().densityDpi
+                / DisplayMetrics.DENSITY_DEFAULT));
+    }
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/util/Util.java]: Both methods seem useless.

File: mobile/src/main/java/org/openhab/habdroid/ui/CloudNotificationAdapter.java [hunk 2]
```diff
@@ -154,7 +155,11 @@ public void bind(CloudNotification notification) {
                 Connection conn = ConnectionFactory.getConnection(Connection.TYPE_CLOUD);
                 String iconUrl = String.format(Locale.US, "images/%s.png",
                         Uri.encode(notification.icon()));
-                mIconView.setImageUrl(conn, iconUrl, 2000);
+                mIconView.setImageUrl(
+                        Util.convertDpToPixel(
+                                mIconView.getResources().getDimension(R.dimen.notificationlist_icon_size),
+                                mIconView.getContext()),
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/CloudNotificationAdapter.java]: `mIconView.setImageUrl(mIconView.getResources.getDimensionPixelSize(R.dimen.notificationlist_icon_size), conn, iconUrl, 2000);`

(`getDimensionPixelSize` returns, unless I'm utterly mistaken, pixel values)

File: mobile/src/main/java/org/openhab/habdroid/ui/widget/WidgetImageView.java [hunk 3]
```diff
@@ -87,24 +87,23 @@ public WidgetImageView(Context context, AttributeSet attrs, int defStyle) {
...
-                context.getResources().getDimensionPixelSize(R.dimen.svg_image_default_size);
+        mDefaultSvgSize = context.getResources().getDimensionPixelSize(R.dimen.svg_image_default_size);
         mRefreshHandler = new RefreshHandler(this);
     }
 
-    public void setImageUrl(Connection connection, String url) {
-        setImageUrl(connection, url, false);
+    public void setImageUrl(Integer size, Connection connection, String url) {
+        setImageUrl(size, connection, url, false);
     }
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/widget/WidgetImageView.java]: I'd prefer putting size as last parameter, to keep parameters ordered by importance.

File: mobile/src/main/java/org/openhab/habdroid/util/AsyncHttpClient.java [hunk 4]
```diff
@@ -78,19 +77,52 @@ public Bitmap convertBodyInBackground(ResponseBody body) throws IOException {
...
-        private Bitmap getBitmapFromSvgInputstream(InputStream is) throws SVGParseException {
+        private static Bitmap getBitmapFromSvgInputstream(Resources res, InputStream is, int size)
+                throws SVGParseException {
             SVG svg = SVG.getFromInputStream(is);
-            RectF viewBox = svg.getDocumentViewBox();
-            double width = viewBox != null ? viewBox.width() : mDefaultSize;
-            double height = viewBox != null ? viewBox.height() : mDefaultSize;
-
+            svg.setRenderDPI(DisplayMetrics.DENSITY_DEFAULT);
+            Float density = res.getDisplayMetrics().density;
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/util/AsyncHttpClient.java]: Float -> float

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/util/AsyncHttpClient.java]: density might be set to null below

File: mobile/src/main/java/org/openhab/habdroid/ui/widget/WidgetImageView.java (hunk 3 above)
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/widget/WidgetImageView.java]: Will do

File: mobile/src/main/java/org/openhab/habdroid/ui/CloudNotificationAdapter.java (hunk 2 above)
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/CloudNotificationAdapter.java]: I didn't know about this function. I'll change it and remove both convert functions in Util.java.

File: mobile/src/main/java/org/openhab/habdroid/util/AsyncHttpClient.java (hunk 4 above)
Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/util/AsyncHttpClient.java]: Oops, right...didn't remember my own code anymore :-/

File: mobile/src/main/java/org/openhab/habdroid/ui/widget/WidgetImageView.java [hunk 5]
```diff
@@ -87,24 +87,23 @@ public WidgetImageView(Context context, AttributeSet attrs, int defStyle) {
             a.recycle();
         }
 
-        mDefaultSvgSize =
-                context.getResources().getDimensionPixelSize(R.dimen.svg_image_default_size);
+        mDefaultSvgSize = context.getResources().getDimensionPixelSize(R.dimen.svg_image_default_size);
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/widget/WidgetImageView.java]: This line is too long, but it's shown as `mDefaultSvgSize  = 60dp;` in AS.

**Pull request merged**